exporter/metric/prometheus/prometheus.go

```diff
@@ -104,70 +92,163 @@ func NewExporter(opts Options) (*Exporter, error) {
 		opts.Gatherer = opts.Registry
 	}
 
-	return &Exporter{
-		registerer:         opts.Registerer,
-		gatherer:           opts.Gatherer,
-		handler:            promhttp.HandlerFor(opts.Gatherer, promhttp.HandlerOpts{}),
-		measureAggregation: opts.MeasureAggregation,
+	if opts.DefaultSummaryObjectives == nil {
+		opts.DefaultSummaryObjectives = map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001}
```

Comment:
Sorry, can you explain this better? this sets me super uneasy because it's (1-N)/10 which feels off?

Reply:
... and then this can be just a []float64 of the map keys here.

Reply:
This was some leftovers from the previous implementation. Since we stopped using Prometheus aggregators, we no longer need an objective here, just quantiles.

Changed this to a slice of floats as Josh suggested.

Reply:
Are 0.5, 0.9 and 0.99 good default values? The behavior in prometheus is not to collect any quantiles if user didn't specify. Not sure what the reason is but it could be that there is no good default value.

Reply:
This was the default from Prometheus but was deprecated. I went through prometheus PR's and [found this PR](https://github.com/prometheus/client_golang/pull/262), the reason seems to be "users shouldn't really be using these in the first place". That feels strange to me, since histogram have default buckets. 

I agree with you that we could defer this decision to the user and simply not set default right now. It might be easier in the future to add these defaults than removing it.

Reply:
Sigh. It's true that exporting local quantile information is not very useful for global aggregation.  Histograms are better in some ways, but having to specify fixed bucket ranges up front makes them not very good either. There's an ongoing question about whether DataDog intends to add support to dogstatsd for recording DDSketch state, which would be a better approach for sending measure data.

Reply:
(The DDSketch library itself doesn't expose public methods for getting its internal, variable histogram buckets. There's no support there even if we wanted to. In the short term, we should just implement histograms.)

Reply:
Maybe we could make the default quantiles be { 0, 1 } which would expose the min/max/sum/count by default. Those quantiles are useful for global aggregation, unlike the values >0 and <1.

Reply:
I'm not too familiar with the datadog and ddsketch. But these quantiles that is being defined here is only used when exporting metrics to Prometheus. So this doesn't actually configure DDSketch directly. 

Even with no quantiles defined, the exporter will keep exporting Sum and Count from measurements.

Maybe I'm misunderstanding what you're suggesting here.

Reply:
You're understanding me, I think. I think just exporting Sum and Count is acceptable. I might advocate for exporting Min (q=0) and Max (q=1) as well, by default, but let's stick with what Prometheus does.

The point about DDSketch is off-topic, related to the "users shouldn't really be using these in the first place" remark above. The quick summary is that to compute global quantiles from a bunch of locally computed quantiles can't be done correctly (except when q=0 or q=1). You can do this with a histogram, but it's only as accurate as the bucket ranges that are configured up front. There are many techniques (like DDSketch) that address this by computing a dynamic histogram, essentially, where the bucket boundaries move according to the data. 

I'm not sure how Prometheus would respond if each client reported different histogram boundaries, but if that's acceptable, then the "correct" thing to do for reporting distributions would be to report a dynamic histogram. DDSketch contains one internally, but doesn't expose getters to access the buckets directly. If this were a goal of ours, we could try to push DDSketch to expose a serialization format (I've reached out to DD about this independently), or we could go with similar approaches (e.g., https://github.com/influxdata/tdigest).

Reply:
I'm resolving this, there's no action needed. If Prometheus uses no quantiles by default in a summary, that's a fine default.

Reply:
Oh, I get it now.

Yeah, I don't think Prometheus would work with a dynamic histogram, we should try to reach some developer from Prometheus to confirm this.